erpnext/assets/doctype/asset/asset.py

```diff
@@ -71,8 +71,19 @@ def validate_asset_values(self):
 		if not flt(self.gross_purchase_amount):
 			frappe.throw(_("Gross Purchase Amount is mandatory"), frappe.MandatoryError)
 
+		if not (self.purchase_receipt or self.purchase_invoice):
```

Comment:
These validations will be only applicable for non-existing assets

Reply:
Fixed!